Dockerfile

```diff
@@ -1,32 +1,30 @@
 FROM alpine:3.6
 
-CMD ["/usr/local/bin/redis-server", "--dir", "/var/lib/redis"]
+ENV RELEASE=4.0-rc3
 
-RUN apk --no-cache --update add \
-        autoconf \
-        g++ \
-        gcc \
-        git \
-        jemalloc-dev \
-        linux-headers \
-        make \
-    && \
-    git clone --branch 4.0-rc3 https://github.com/antirez/redis.git /tmp/redis && \
+WORKDIR /var/lib/redis
+
+RUN apk --no-cache add \
+        --virtual build-dependencies \
+          autoconf \
+          g++ \
+          gcc \
+          git \
+          jemalloc-dev \
+          linux-headers \
+          make && \
+    git clone --depth=1 --branch ${RELEASE} https://github.com/antirez/redis.git /tmp/redis/&& \
     cd /tmp/redis && \
-    make && \
+    make -j && \
     make install && \
     cp /tmp/redis/src/redis-trib.rb /usr/local/bin && \
-    rm -rf /tmp/* && \
-    apk del \
-        autoconf \
-        g++ \
-        gcc \
-        git \
-        jemalloc-dev \
-        linux-headers \
-        make \
-    && \
-    mkdir /var/lib/redis && \
-    chown guest /var/lib/redis
+    rm -rf /tmp/redis && \
+    chown guest /var/lib/redis && \
+    apk del build-dependencies
 
 USER guest
+
+EXPOSE 6379
+
+ENTRYPOINT ["/usr/local/bin/redis-server"]
+CMD ["--dir", "/var/lib/redis"]
```

Comment:
These, along with expose, should be at the top of the Dockerfile to optimize for cache layers on rebuild.